crates/rv/src/commands/ruby/install.rs

```diff
@@ -111,51 +108,44 @@ fn temp_tarball_path(config: &Config, url: impl AsRef<str>) -> Utf8PathBuf {
         .join(format!("{cache_key}.tar.gz.tmp"))
 }
 
+/// Write the file from this HTTP `response` to the given `path`.
+/// While the stream is being handled, it'll be written to the given `temp_path`.
+/// Then once the download finishes, the file will be renamed to `path`.
+async fn write_to_filesystem(
+    response: reqwest::Response,
+    temp_path: &Utf8Path,
+    path: &Utf8Path,
+) -> Result<()> {
+    let mut file = tokio::fs::File::create(&temp_path).await?;
```

Comment:
On main, the file is created before the download starts. By moving the temp file creation here, we ensure there's no empty tempfile to clean up if the download fails (e.g. because of a 404)